src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import commands.AddCommand;
+import commands.Command;
+import commands.DeleteCommand;
+import commands.DoneCommand;
+import commands.ExitCommand;
+import commands.FindCommand;
+import commands.HelpCommand;
+import commands.ListCommand;
+import commands.UndoneCommand;
+
+
+/**
+ * Handles parsing of inputs from the user and generates Command objects accordingly.
+ */
+public class Parser {
+
+    private boolean isQuit = false;
+
+    public boolean getQuit() {
+        return isQuit;
+    }
```

Comment:
I understand that you avoid isQuit again because it has the same naming as the attribute. Perhaps can consider hasQuit() so it is clearer that its a boolean?